tests/foundry/test/UintTypeTest.t.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.13;
+
+contract UintTypeTest {
+    function setUp() public {}
+
+    function test_uint256(uint256 x) public pure {
+        assert(x == x);
+        assert(type(uint256).max >= x);
+    }
+
+    function testFail_uint256(uint256 x) public pure {
+        assert(type(uint256).max > x);
+    }
+
+    function test_uint224(uint224 x) public pure {
+        assert(x == x);
+        assert(type(uint224).max >= x);
+    }
+
+    function testFail_uint224(uint224 x) public pure {
+        assert(type(uint224).max > x);
+    }
+
+    function test_uint160(uint160 x) public pure {
+        assert(x == x);
+        assert(type(uint160).max >= x);
+    }
+
+    function testFail_uint160(uint160 x) public pure {
+        assert(type(uint160).max > x);
+    }
+
+    function test_uint144(uint144 x) public pure {
+        assert(x == x);
+        assert(type(uint144).max >= x);
+    }
+
+    function testFail_uint144(uint144 x) public pure {
+        assert(type(uint144).max > x);
+    }
+
+    function test_uint112(uint112 x) public pure {
+        assert(x == x);
```

Comment:
What is the reason to have these `assert(x == x)`?

Reply:
To use `x` for something, rather than just being `assert(true)`. Actually, these tests passed with just `assert(true)` before making any changes to `lemmas.k`

Reply:
The point of the first assert is to just use the variable, and the second assert is to test out `type(uintX).max` for something. I did it like this so that it's easier to debug/isolate what "components" of the types we're testing. For instance, I made a typo on one of the `type(uintX).max` and the first assert was passing but not the second. But maybe we want to subsume these into just one assert.

Reply:
Ahh ok makes sense! Thanks! Other alternative is to put them into two different tests, one to test just the type with the first `assert` and other to test `type(uintX).max` with the second `assert`.

Reply:
After thinking about this, since both `assert`s are reasonably trivial, I'd leave them together so that the time to build and execute the tests doesn't increase too much. What do you think about this?

Reply:
You are right! It's better to leave them together.